src/codec_aom.c

```diff
@@ -703,103 +712,84 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
                 cfg->monochrome = 1;
             }
         }
-    }
-
-    avifBool dimensionsChanged = AVIF_FALSE;
-    if (!codec->internal->encoderInitialized) {
-        cfg->g_w = image->width;
-        cfg->g_h = image->height;
-    } else if ((cfg->g_w != image->width) || (cfg->g_h != image->height)) {
-        // We are not ready for dimension change for now.
-        return AVIF_RESULT_NOT_IMPLEMENTED;
-    }
 
-    if (!codec->internal->encoderInitialized || encoderChanges) {
-        int minQuantizer = AVIF_CLAMP(encoder->minQuantizer, 0, 63);
-        int maxQuantizer = AVIF_CLAMP(encoder->maxQuantizer, 0, 63);
-        if (alpha) {
-            minQuantizer = AVIF_CLAMP(encoder->minQuantizerAlpha, 0, 63);
-            maxQuantizer = AVIF_CLAMP(encoder->maxQuantizerAlpha, 0, 63);
+        if (!avifProcessAOMOptionsPreInit(codec, alpha, cfg)) {
+            return AVIF_RESULT_INVALID_CODEC_SPECIFIC_OPTION;
         }
-        lossless = ((minQuantizer == AVIF_QUANTIZER_LOSSLESS) && (maxQuantizer == AVIF_QUANTIZER_LOSSLESS));
-        cfg->rc_min_quantizer = minQuantizer;
-        cfg->rc_max_quantizer = maxQuantizer;
-    }
 
-    if (!codec->internal->encoderInitialized) {
         aom_codec_flags_t encoderFlags = 0;
         if (image->depth > 8) {
             encoderFlags |= AOM_CODEC_USE_HIGHBITDEPTH;
         }
-
-        if (!avifProcessAOMOptionsPreInit(codec, alpha, cfg)) {
-            return AVIF_RESULT_INVALID_CODEC_SPECIFIC_OPTION;
-        }
-
         if (aom_codec_enc_init(&codec->internal->encoder, encoderInterface, cfg, encoderFlags) != AOM_CODEC_OK) {
             avifDiagnosticsPrintf(codec->diag,
                                   "aom_codec_enc_init() failed: %s: %s",
                                   aom_codec_error(&codec->internal->encoder),
                                   aom_codec_error_detail(&codec->internal->encoder));
             return AVIF_RESULT_UNKNOWN_ERROR;
         }
+        codec->internal->encoderInitialized = AVIF_TRUE;
```

Comment:
Note that we must set `codec->internal->encoderInitialized` to true immediately after a successful `aom_codec_enc_init()` call, because we also check `codec->internal->encoderInitialized` to determine if `aom_codec_destroy(codec->internal->encoder)` should be called:

```
#if defined(AVIF_CODEC_AOM_ENCODE)
    if (codec->internal->encoderInitialized) {
        aom_codec_destroy(&codec->internal->encoder);
    }
#endif
```

Perhaps this should be documented better.